armory/art_experimental/attacks/carla_obj_det_adversarial_patch.py

```diff
@@ -66,51 +85,47 @@ def _train_step(
+                    depth_grads[:, 0, :, :],
+                    depth_grads[:, 1, :, :],
+                    depth_grads[:, 2, :, :],
                 )
-
-            gradients = patch_grads.sign() * self.learning_rate
-            if patch_grads.shape[0] == 6:
-                gradients[3:, :, :] = (
-                    gradients[3:, :, :] / self.learning_rate * self.learning_rate_depth
+                depth_linear = rgb_depth_to_linear(
+                    self.depth_perturbation[:, 0, :, :],
+                    self.depth_perturbation[:, 1, :, :],
+                    self.depth_perturbation[:, 2, :, :],
+                )
+                depth_linear = (
+                    depth_linear + grads_linear.sign() * self.learning_rate_depth
                 )
 
             with torch.no_grad():
                 self._patch[:] = torch.clamp(
-                    self._patch + gradients,
+                    self._patch + patch_gradients,
                     min=self.estimator.clip_values[0],
                     max=self.estimator.clip_values[1],
                 )
 
-                if self._patch.shape[0] == 6:
-                    if self.depth_type == "linear":
-                        self._patch[3, :, :] = torch.clamp(
-                            self._patch[3, :, :],
-                            min=self.min_depth_r,
-                            max=self.max_depth_r,
-                        )
-                        self._patch[4, :, :] = torch.clamp(
-                            self._patch[4, :, :],
-                            min=self.min_depth_g,
-                            max=self.max_depth_g,
-                        )
-                        self._patch[5, :, :] = torch.clamp(
-                            self._patch[5, :, :],
-                            min=self.min_depth_b,
-                            max=self.max_depth_b,
-                        )
-                    elif self.depth_type == "log":
-                        self._patch[3:, :, :] = torch.clamp(
-                            self._patch[3:, :, :],
-                            min=self.min_depth,
-                            max=self.max_depth,
-                        )
-                    else:
-                        raise ValueError(
-                            f"Expected depth_type in ('log', 'linear'). Found {self.depth_type}"
-                        )
+                if images.shape[-1] == 6:
+                    images_depth = torch.permute(images[:, :, :, 3:], (0, 3, 1, 2))
+                    images_depth_linear = rgb_depth_to_linear(
+                        images_depth[:, 0, :, :],
+                        images_depth[:, 1, :, :],
+                        images_depth[:, 2, :, :],
+                    )
+                    depth_linear = torch.clamp(
+                        images_depth_linear + depth_linear,
+                        min=self.min_depth,
+                        max=self.max_depth,
+                    )
```

Comment:
```suggestion
                    if depth_linear.shape != self.min_depth.shape:
                        assert torch.diff(self.min_depth, dim=-1).sum() == 0
                        assert torch.diff(self.max_depth, dim=-1).sum() == 0
                        depth_linear = torch.clamp(
                            images_depth_linear + depth_linear,
                            min=self.min_depth[..., 0],
                            max=self.max_depth[..., 0],
                        )
                    else:
                        depth_linear = torch.clamp(
                            images_depth_linear + depth_linear,
                            min=self.min_depth,
                            max=self.max_depth,
                        )
```

Reply:
@jprokos26 my latest commit fixes the issue.